File: src/main/java/io/gittul/domain/comment/dto/CommentResponse.java
package io.gittul.domain.comment.dto;

import io.gittul.domain.comment.entity.Comment;
import io.gittul.domain.user.dto.UserProfileResponse;
import io.gittul.domain.user.entity.User;

import java.util.ArrayList;
import java.util.List;

public record CommentResponse(
        UserProfileResponse user,

        String content,

        String createdAt,

        String updatedAt,

        int likeCount,

        Long id,

        boolean isLiked,

        List<CommentResponse> replies
) {
    public static CommentResponse ofAndTo(Comment comment,
                                          User requestingUser) {
        return new CommentResponse(
                UserProfileResponse.of(comment.getUser()),
                comment.getContent(),
                comment.getCreatedAt().toString(),
                comment.getUpdatedAt().toString(),
                comment.getLikeCount(),
                comment.getCommentId(),
                comment.isLikedBy(requestingUser),
                new ArrayList<>() // Todo. 대댓글 구현
        );
    }
}

File: app/src/main/kotlin/io/gittul/app/domain/comment/dto/CommentResponse.kt
package io.gittul.app.domain.comment.dto

import io.gittul.app.domain.user.dto.UserProfileResponse
import io.gittul.core.domain.comment.entity.Comment
import io.gittul.core.domain.user.entity.User

data class CommentResponse(
    val user: UserProfileResponse?,
    val content: String,
    val createdAt: String,
    val updatedAt: String,
    val likeCount: Int,
    val id: Long,
    val isLiked: Boolean,
    val replies: List<CommentResponse?>
) {
    companion object {
        fun ofAndTo(comment: Comment?, requestingUser: User?): CommentResponse? {
            if (comment == null) return null

            return CommentResponse(
                user = UserProfileResponse.of(comment.user),
                content = comment.content ?: "",
                createdAt = comment.createdAt?.toString() ?: "",
                updatedAt = comment.updatedAt?.toString() ?: "",
                likeCount = comment.likeCount,
                id = comment.commentId,
                isLiked = comment.isLikedBy(requestingUser),
                replies = emptyList()  // Todo. 대댓글 추가
            )
        }
    }
}
